grammar.js

```diff
@@ -312,14 +313,13 @@ module.exports = grammar({
     for_statement: $ => seq(
       'for',
       '(',
-      field('initializer', choice(
-        $.lexical_declaration,
-        $.variable_declaration,
-        $.expression_statement,
-        $.empty_statement,
-      )),
+      choice(
+        field('initializer', choice($.lexical_declaration, $.variable_declaration)),
+        seq(field('initializer', $._expressions), ';'),
+        field('initializer', $.empty_statement),
+      ),
       field('condition', choice(
-        $.expression_statement,
+        seq($._expressions, ';'),
```

Comment:
hmm, do you mind creating a `_expression_statement_no_auto_semi` node or something similar, and then alias that here to `expression_statement`? this way querying for these nodes works the same as before, and all `expression_statement` nodes can be captured the same as before.

Reply:
I was against doing that because both the [ESTree spec](https://github.com/estree/estree/blob/master/es5.md#forstatement) and the [ecmascript spec](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-for-statement) consider the clauses of the for statement to be expressions and not statements.

I get your point though that this would be backwards-incompatible. Do you think maintaining the backwards-compatibility is worth the deviation from the norm here?

Reply:
that's totally fine then